packages/main/src/ColorPalette.ts

```diff
@@ -241,26 +244,60 @@ class ColorPalette extends UI5Element {
 		});
 	}
 
+	/**
+	 * Ensures that only one item is selected or only the last selected item remains active if more than one are explicitly set as 'selected'.
+	 *
+	 * @param {boolean} deselectAll - If true, will deselect all items. Otherwise, ensures only the last selected item remains active.
+	 * @private
+	 * @returns {void}
+	 */
+	_ensureSingleSelectionOrDeselectAll(deselectAll = false) {
+		const selectedItems = [...this.colors, ...this.recentColorsElements].filter(item => item.selected);
+		!deselectAll && selectedItems.pop();
+		selectedItems.forEach(item => { item.selected = false; });
+	}
+
 	_onclick(e: MouseEvent) {
-		const target = e.target as ColorPaletteItem;
-		if (target.hasAttribute("ui5-color-palette-item")) {
-			this.selectColor(target);
-		}
+		this.handleSelection(e.target as ColorPaletteItem);
 	}
 
 	_onkeyup(e: KeyboardEvent) {
 		const target = e.target as ColorPaletteItem;
-		if (isSpace(e) && target.hasAttribute("ui5-color-palette-item")) {
+		if (isSpace(e)) {
 			e.preventDefault();
-			this.selectColor(target);
+			this.handleSelection(target);
 		}
 	}
 
 	_onkeydown(e: KeyboardEvent) {
 		const target = e.target as ColorPaletteItem;
-		if (isEnter(e) && target.hasAttribute("ui5-color-palette-item")) {
-			this.selectColor(target);
+		if (isEnter(e)) {
+			this.handleSelection(target);
+		}
+	}
+
+	handleSelection(target: ColorPaletteItem) {
```

Comment:
With this refactored function we face two problems:

#### 1. In the `Recent Colors`, when selecting an item, since they _shift_, the selected item becomes the item **before** the desired item to be selected, and the item you wanted to select, gets shifted to the first place of the Recent Colors. 

Please refer:

https://github.com/SAP/ui5-webcomponents/assets/88034608/e7cfca41-2b9a-485c-8f3d-52803d9831ec

Which is addressed by this part of the current implementation:

```ts
if (this.recentColorsElements.includes(target)) {
	target.selected = true;
	this._selectedColor = target.value;
	return;
}
```

In my opinion lets leave it as it is, but just add deselecting functionality to the Recent Colors as in the ColorPalette ? What do you think

#### 2. Once item is selected and then deselected, it cannot be selected again.

https://github.com/SAP/ui5-webcomponents/assets/88034608/cf7f34d2-86d8-40ea-8a0a-d0988894509f

Reply:
1. It works as expected or the behaviour in OpenUI5 is also wrong
2. We can fix that by simply changing this code:

```ts
### Before
if (this._currentlySelected === target) {
	target.selected = false;
	return;
}

### After
if (this._currentlySelected === target) {
	target.selected = !target.selected;
	return;
}
```

Reply:
For `_ensureSingleSelectionOrDeselectAll` method just remove the `deselectAll` parameter
```ts
const selectedItems = [...this.colors, ...this.recentColorsElements].filter(item => item.selected);

selectedItems.pop();

selectedItems.forEach(item => { item.selected = false; });
```